engines/groovie/t11hgame.cpp

```diff
@@ -0,0 +1,234 @@
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*
+*/
+
+#include "groovie/t11hgame.h"
+#include "groovie/groovie.h"
+
+#include "common/archive.h"
+#include "common/config-manager.h"
+#include "common/debug-channels.h"
+#include "common/events.h"
+#include "common/file.h"
+#include "common/macresman.h"
+#include "common/translation.h"
+
+namespace Groovie {
+
+/* Links between the pieces in the Gallery challenge */
+byte kGalleryLinks[21][10] = {
+	{ 2,  4,  5,  0,  0,  0,  0,  0,  0,  0 },	//  1
+	{ 1,  5,  3,  0,  0,  0,  0,  0,  0,  0 },	//  2
+	{ 2,  5,  9, 12,  0,  0,  0,  0,  0,  0 },	//  3
+	{ 1,  5,  6,  7,  8,  0,  0,  0,  0,  0 },	//  4
+	{ 1,  2,  3,  4,  7,  8,  9,  0,  0,  0 },	//  5
+	{ 4,  7, 10, 11, 13, 14, 15, 16, 18,  0 },	//  6
+	{ 4,  5,  6,  8,  9, 10,  0,  0,  0,  0 },	//  7
+	{ 4,  5,  7,  0,  0,  0,  0,  0,  0,  0 },	//  8
+	{ 3,  5,  7, 10, 11, 12, 18,  0,  0,  0 },	//  9
+	{ 6,  7,  9, 11,  0,  0,  0,  0,  0,  0 },	// 10
+	{ 6,  9, 10, 18,  0,  0,  0,  0,  0,  0 },	// 11
+	{ 3,  9, 18, 21,  0,  0,  0,  0,  0,  0 },	// 12
+	{ 6, 14, 17, 19,  0,  0,  0,  0,  0,  0 },	// 13
+	{ 6, 13, 15, 17, 19, 20, 21,  0,  0,  0 },	// 14
+	{ 6, 14, 16, 18, 21,  0,  0,  0,  0,  0 },	// 15
+	{ 6, 15,  0,  0,  0,  0,  0,  0,  0,  0 },	// 16
+	{13, 14, 19,  0,  0,  0,  0,  0,  0,  0 },	// 17
+	{ 6,  9, 11, 12, 15, 21,  0,  0,  0,  0 },	// 18
+	{13, 14, 17, 20,  0,  0,  0,  0,  0,  0 },	// 19
+	{14, 19, 21,  0,  0,  0,  0,  0,  0,  0 },	// 20
+	{12, 14, 15, 18, 20,  0,  0,  0,  0,  0 }	// 21
+};
+
+T11hGame::T11hGame() :
+	_random("GroovieT11hGame"), _scriptVariables(NULL) {
+}
+
+
+T11hGame::~T11hGame() {
+}
+
+
+void T11hGame::setVariables(byte *scriptVariables) {
+	_scriptVariables = scriptVariables;
+}
+
+
+/*
+* Mouse Trap puzzle in the Lab.
+*
+* Stauf's Goal is space 1, counting up as you go north east towards the north corner which is space 5 and the moveable space to the left of that is space 4
+* South east from Stauf's goal is the next line starting with space 6, counting up as you go north east where the moveable space to the right of the north corner is space 10
+* Next line is 11 (unmovable) to 15 (unmoveable), this line contains the center space which is space 13
+* Next line is 16 (moveable) to 20 (moveable)
+* Next line is 21 (unmovable) to 25 (unmovable), with 25 being the player's goal door
+*
+* Space -2 is the next piece, outside of the box
+*/
+void T11hGame::opMouseTrap() {
+	// TODO: properly implement mouse trap game
+	// variable 24 is the mouse?
+	//_scriptVariables[24] = 2;
+
+	// value of 0 is V, 1 is <, 2 is ^, 3 is >
+	// variable 23 is the outside piece
+	_scriptVariables[23] = 1;
+	// variable slot is the space number + 25, the left corner (Stauf's goal) is space 1, above that is space 2, the center is 13, and the right corner (goal) is space 25
+	for(int i=27; i<=49; i++)

```

Comment:
Spacing. Please check our code formatting guidelines in the wiki

```suggestion
	for (int i = 27; i <= 49; i++)
```

Reply:
ok sorry, visual studio was doing these for me automatically so I guess it missed that part